app/main.py

```diff
@@ -1,3 +1,63 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, add_: int | Distance) -> Distance:
+        if isinstance(add_, (int, float)):
+            return Distance(self.km + add_)
+        return Distance(self.km + add_.km)
+
+    def __iadd__(self, iadd_: int | Distance) -> Distance:
+        if isinstance(iadd_, (int, float)):
+            self.km += iadd_
+            return self
+        else:
+            self.km += iadd_.km
+            return self
+
+    def __mul__(self, mul_: int) -> Distance:
```

Comment:
type of mul_ can be `float` also . Check in other places where argument can be `int` and `float`